lib/AST/Decl.cpp

```diff
@@ -5945,12 +5945,23 @@ EnumCaseDecl *EnumCaseDecl::create(SourceLoc CaseLoc,
 
 bool EnumDecl::hasPotentiallyUnavailableCaseValue() const {
   switch (static_cast<AssociatedValueCheck>(Bits.EnumDecl.HasAssociatedValues)) {
-    case AssociatedValueCheck::Unchecked:
-      // Compute below
-      this->hasOnlyCasesWithoutAssociatedValues();
-      LLVM_FALLTHROUGH;
-    default:
-      return static_cast<bool>(Bits.EnumDecl.HasAnyUnavailableValues);
+  case AssociatedValueCheck::Unchecked:
+    this->hasOnlyCasesWithoutAssociatedValues();
+    LLVM_FALLTHROUGH;
+  default:
+    return static_cast<bool>(Bits.EnumDecl.HasAnyUnavailableValues);
+  }
+}
+
+bool EnumDecl::hasCasesUnavailableDuringLowering() const {
+  switch (
+      static_cast<AssociatedValueCheck>(Bits.EnumDecl.HasAssociatedValues)) {
```

Comment:
Rather than have this large thing... maybe add a helper method that encapsulates the cast? Your choice though.